File: string_to_int.py
'''Given a number represented as a string convert it to an integer without using "int()"'''


def string_to_int(input_string):
    output = 0
    int_dict = {"0": 0, "1": 1, "2": 2, "3": 3, "4": 4, "5": 5, "6": 6, "7": 7, "8": 8, "9": 9}
    i = 0
    for char in reversed(input_string):
        print "char", char
        index_char = i
        print "index", index_char
        digit = int_dict[char]
        print digit
        print "final:", digit * (10 ** index_char)
        output += (digit * (10 ** index_char))
        i += 1
    return output


print string_to_int('132')
print type(string_to_int('132'))

File: string_to_int.rb
'''Given a number represented as a string convert it to an integer without using "int()"'''

DIGITS_MAP = {
    "0" => 0,
    "1" => 1,
    "2" => 2,
    "3" => 3,
    "4" => 4,
    "5" => 5,
    "6" => 6,
    "7" => 7,
    "8" => 8,
    "9" => 9
}

def string_to_int(n)
    output = 0

    n.reverse!.split("").each_with_index do |digit, index|
        digit_int = DIGITS_MAP[digit]
        output += (digit_int * (10 ** index))
    end

    output
end

puts(string_to_int("123"))
puts(string_to_int("123").class)
puts(string_to_int("1000"))
puts(string_to_int("1000").class)